File: app/src/main/java/grey/smarthouse/ui/fragments/SensorDialog.java
package grey.smarthouse.ui.fragments;

import android.annotation.TargetApi;
//import android.app.AlertDialog;
import androidx.appcompat.app.AlertDialog;
import android.app.Dialog;
import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
//import android.support.v7.app.AppCompatDialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;
import android.content.DialogInterface.OnClickListener;
import androidx.annotation.Nullable;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;

import grey.smarthouse.R;

public class SensorDialog extends DialogFragment implements OnClickListener {


    @NonNull
    @Override
    @TargetApi(11)
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        LayoutInflater inflater = getActivity().getLayoutInflater();
        View v =inflater.inflate(R.layout.fragment_dialog_sensor,null);
        return new AlertDialog.Builder(getActivity())
                .setView(v)
                .setTitle(R.string.sensor)
                .setPositiveButton(android.R.string.ok, this)
                .setNegativeButton(R.string.cancel, this)
                .create();

    }

    @Override
    public void onDismiss(DialogInterface dialog) {
        super.onDismiss(dialog);
    }

    @Override
    public void onCancel(DialogInterface dialog) {
        super.onCancel(dialog);
        Log.d("dialog", "Dialog 1: onCancel");
    }

    @Override
    public void onClick(DialogInterface dialog, int which) {
        dismiss();
    }

}

File: app/src/main/java/grey/smarthouse/ui/mainScreen/sensors/SensorDialog.kt
package grey.smarthouse.ui.mainScreen.sensors

import android.annotation.TargetApi
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import androidx.lifecycle.observe
import grey.smarthouse.App
import grey.smarthouse.R
import grey.smarthouse.databinding.FragmentDialogSensorBinding
import javax.inject.Inject

class SensorDialog : DialogFragment() {

    @Inject
    lateinit var sensorsVM: SensorsVM

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        sensorsVM.dialogDismiss.observe(viewLifecycleOwner) {
            it?.let {
                dialog?.dismiss()
            }
        }
    }

    @TargetApi(11)
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val inflater = requireActivity().layoutInflater
        val binding = FragmentDialogSensorBinding.inflate(inflater)
        sensorsVM = (requireActivity().application as App).appComponent.getSensorsVM()
        binding.viewModel = sensorsVM
        binding.lifecycleOwner = requireActivity()
        val sensorId = arguments?.getInt("num") ?: -1
        sensorsVM.fillDescription(sensorId)
        val dialog = AlertDialog.Builder(requireActivity())
            .setView(binding.root)
            .setTitle(R.string.sensor)
            .setPositiveButton(android.R.string.ok, null)
            .setNegativeButton(R.string.cancel, null)
            .create()
        dialog.setOnShowListener {
            dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
                sensorsVM.positiveAction()
            }
        }
        return dialog
    }

    override fun onDismiss(dialog: DialogInterface) {
        sensorsVM.dismissListener()
        super.onDismiss(dialog)
    }
}
